### ISSUE TITLE
The player changed from version 2.0.4 to 2.0.5 and now full screen is not working on iOS

### ISSUE THREAD

**Issue by mvolpato** [SmallDreams/youtube_plyr_iframe#32]
Hi,

I noticed a big difference in the player on iOS between 2.0.4 and 2.0.5

## 2.0.4
<img src="https://user-images.githubusercontent.com/7138011/120105102-cc5e9c00-c157-11eb-9f71-8db2c51b6920.png" width="300" />

## 2.0.5
<img src="https://user-images.githubusercontent.com/7138011/120105107-d4b6d700-c157-11eb-8baa-85d98f41132c.png" width="300" />

Am I missing some change in parameters I need to set?

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.2.0, on macOS 11.3 20E232
    darwin-x64, locale en-GB)
[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from:
      https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the
      Android SDK components.
      (or visit
      https://flutter.dev/docs/get-started/install/macos#an
      droid-setup for detailed instructions).
      If the Android SDK has been installed to a custom
      location, please use
      `flutter config --android-sdk` to update to that
      location.

[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[!] Android Studio (not installed)
[✓] VS Code (version 1.56.2)
[✓] Connected device (3 available)
```

I am using the simulator.

**Comment by jonatadashi** [comment]
Thanks for reporting, seems like I messed up somewhere. Will look into it, asap.

**Comment by jonatadashi** [comment]
What are the parameters you're passing?

**Comment by mvolpato** [comment]
I am using the example (an [old version](https://github.com/SmallDreams/youtube_plyr_iframe/blob/fa4537ed99366aa37613f1a081b3209520db6cb2/example/lib/main.dart#L71)) from the project.

```
// Copyright 2020 Sarbagya Dhaubanjar. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'dart:developer';

import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:youtube_plyr_iframe/youtube_plyr_iframe.dart';

import 'widgets/meta_data_section.dart';
import 'widgets/play_pause_button_bar.dart';
import 'widgets/player_state_section.dart';
import 'widgets/source_input_section.dart';
import 'widgets/volume_slider.dart';

void main() {
  WidgetsFlutterBinding.ensureInitialized();
  runApp(YoutubeApp());
}

///
class YoutubeApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Youtube Plyr Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        iconTheme: const IconThemeData(color: Colors.blue),
      ),
      debugShowCheckedModeBanner: false,
      home: YoutubeAppDemo(),
    );
  }
}

///
class YoutubeAppDemo extends StatefulWidget {
  @override
  _YoutubeAppDemoState createState() => _YoutubeAppDemoState();
}

class _YoutubeAppDemoState extends State<YoutubeAppDemo> {
  late YoutubePlayerController _controller;

  @override
  void initState() {
    super.initState();
    _controller = YoutubePlayerController(
      initialVideoId: '5qap5aO4i9A', // livestream example
      params: YoutubePlayerParams(
        playlist: [
          'F1B9Fk_SgI0',
          "MnrJzXM7a6o",
          "FTQbiNvZqaY",
          "iYKXdt0LRs8",
        ],
        //startAt: Duration(minutes: 1, seconds: 5),
        showControls: true,
        showFullscreenButton: true,
        desktopMode: false, // false for platform design
        autoPlay: false,
        enableCaption: true,
        showVideoAnnotations: false,
        enableJavaScript: true,
        privacyEnhanced: true,
        playsInline: true, // iOS only - Auto fullscreen or not
      ),
    )..listen((value) {
        if (value.isReady && !value.hasPlayed) {
          _controller
            ..hidePauseOverlay()
            // Uncomment below to start autoplay on iOS
            //..play()
            ..hideTopMenu();
        }
      });
    //Uncomment below for auto rotation on fullscreen
    // _controller.onEnterFullscreen = () {
    //   SystemChrome.setPreferredOrientations([
    //     DeviceOrientation.landscapeLeft,
    //     DeviceOrientation.landscapeRight,
    //   ]);
    //   log('Entered Fullscreen');
    // };
    // _controller.onExitFullscreen = () {
    //   SystemChrome.setPreferredOrientations([DeviceOrientation.portraitUp]);
    //   Future.delayed(const Duration(seconds: 1), () {
    //     _controller.play();
    //   });
    //   Future.delayed(const Duration(seconds: 5), () {
    //     SystemChrome.setPreferredOrientations(DeviceOrientation.values);
    //   });
    //   log('Exited Fullscreen');
    // };
  }

  @override
  Widget build(BuildContext context) {
    const player = YoutubePlayerIFrame();
    return YoutubePlayerControllerProvider(
      // Passing controller to widgets below.
      controller: _controller,
      child: Scaffold(
        appBar: AppBar(
          title: const Text('Youtube Plyr Demo'),
        ),
        body: LayoutBuilder(
          builder: (context, constraints) {
            if (kIsWeb && constraints.maxWidth > 800) {
              return Row(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  const Expanded(child: player),
                  const SizedBox(
                    width: 500,
                    child: SingleChildScrollView(
                      child: Controls(),
                    ),
                  ),
                ],
              );
            }
            return ListView(
              children: [
                player,
                const Controls(),
              ],
            );
          },
        ),
      ),
    );
  }

  @override
  void dispose() {
    _controller.close();
    super.dispose();
  }
}

///
class Controls extends StatelessWidget {
  ///
  const Controls();

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.all(16),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          _space,
          MetaDataSection(),
          _space,
          SourceInputSection(),
          _space,
          PlayPauseButtonBar(),
          _space,
          VolumeSlider(),
          _space,
          PlayerStateSection(),
        ],
      ),
    );
  }

  Widget get _space => const SizedBox(height: 10);
}
```

Let me try the latest version, I will update you.

**Comment by mvolpato** [comment]
I used the latest example (`oldDemo.dart`) and now it seems to work according to the `desktopMode` parameter. If I use `desktopMode: true` then the full screen button does not work (it shows the error it shows in the image above. 

I am fine with this. You can close, if that is the correct behaviour.

**Comment by jonatadashi** [comment]
Ah, I see where I went wrong.
https://github.com/SmallDreams/youtube_plyr_iframe/blob/1bba95fee2d40b747f7e2fc5fc47e0731cba4e66/lib/src/players/youtube_player_mobile.dart#L349

Yeah `desktopMode: false` would be what it said, somehow I messed that up and made a reverse uno, which shouldn't be.

Fixed with `youtube_plyr_iframe: ^2.0.6`
